common/thorhelper/thorstrand.hpp

```diff
@@ -0,0 +1,186 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef THORSTRAND_HPP
+#define THORSTRAND_HPP
+
+#include "jqueue.hpp"
+#include "thorhelper.hpp"
+#include "roxiestream.hpp"
+#include "roxiemem.hpp"
+
+class IStrandJunction : public IInterface
+{
+public:
+    virtual IEngineRowStream * queryOutput(unsigned n) = 0;
+    virtual void setInput(unsigned n, IEngineRowStream * _stream) = 0;
+    virtual void stop() = 0; // input has finished reading
+    virtual void ready() = 0;
+    virtual void reset() = 0;
+    virtual void abort() = 0;
+};
+
+interface IRowWriterEx : public IRowWriter
+{
+public:
+    virtual void noteStopped() = 0;
+};
+
+interface IManyToOneRowSteam : public IRowStream
+{
+public:
+    virtual IRowWriterEx * getWriter(unsigned n) = 0;
+    virtual void abort() = 0;
+};
+
+interface IStrandBranch : public IInterface
```

Comment:
Is the consensus that interface is good, but 'extends' is not ?
I still use extends ..